sql/updates/1.7.1--1.7.2.sql

```diff
@@ -16,59 +16,66 @@ WITH
    -- All dimension slices that are mentioned in the chunk_constraint
    -- table but are missing from the dimension_slice table.
    missing_slices AS (
-      SELECT dimension_slice_id,
-      	     constraint_name,
-	     attname AS column_name,
-	     pg_get_expr(conbin, conrelid) AS constraint_expr
+      SELECT ch.hypertable_id,
+             di.id as dimension_id,
+             dimension_slice_id,
+             constraint_name,
+             di.column_type,
+             attname AS column_name,
+             pg_get_expr(conbin, conrelid) AS constraint_expr
       FROM _timescaledb_catalog.chunk_constraint cc
       JOIN _timescaledb_catalog.chunk ch ON cc.chunk_id = ch.id
       JOIN pg_constraint ON conname = constraint_name
       JOIN pg_namespace ns ON connamespace = ns.oid AND ns.nspname = ch.schema_name
       JOIN pg_attribute ON attnum = conkey[1] AND attrelid = conrelid
+      JOIN _timescaledb_catalog.dimension di
+           ON ch.hypertable_id = di.hypertable_id AND di.column_name = attname
       WHERE
-	 dimension_slice_id NOT IN (SELECT id FROM _timescaledb_catalog.dimension_slice)
+         dimension_slice_id NOT IN (SELECT id FROM _timescaledb_catalog.dimension_slice)
    ),
 
   -- Unparsed range start and end for each dimension slice id that
   -- is missing.
    unparsed_missing_slices AS (
-      SELECT dimension_slice_id,
+      SELECT dimension_id,
+             dimension_slice_id,
              constraint_name,
-	     column_name,
-	     (SELECT SUBSTRING(constraint_expr, $$>=\s*'?([\d\s:+-]+)'?$$)) AS range_start,
-	     (SELECT SUBSTRING(constraint_expr, $$<\s*'?([\d\s:+-]+)'?$$)) AS range_end
-	FROM missing_slices
+             column_type,
+             column_name,
+             (SELECT SUBSTRING(constraint_expr, $$>=\s*'?([\d\s:+-]+)'?$$)) AS range_start,
+             (SELECT SUBSTRING(constraint_expr, $$<\s*'?([\d\s:+-]+)'?$$)) AS range_end
+        FROM missing_slices
    )
-SELECT dimension_slice_id,
-       di.id AS dimension_id,
+SELECT DISTINCT dimension_slice_id,
```

Comment:
When can duplicates appear? Is it due to calculated range start and end? Or do they produced earlier, e.g., in `missing_slices`?

Reply:
There can be several chunks that refer to a dimension slice, which will generate duplicates in the select, so they are, sort of, "produced" by the `missing_slices`.

Reply:
Thank you for explaining. Can `DISTINCT` be moved to the missing slices query?

Reply:
It seems to me there should not be any obstacles to move `DISTINCT` to the query producing `missing_slices`. This will make semantics of duplicate appearance more clear.